x/evm/types/msg.go

```diff
@@ -242,18 +254,50 @@ func (msg MsgEthereumTx) GetSignBytes() []byte {
 func (msg MsgEthereumTx) RLPSignBytes(chainID *big.Int) ethcmn.Hash {
 	return rlpHash([]interface{}{
 		msg.Data.AccountNonce,
-		msg.Data.Price,
+		msg.Data.Price.BigInt(),
 		msg.Data.GasLimit,
-		msg.Data.Recipient,
-		msg.Data.Amount,
+		msg.To(),
+		msg.Data.Amount.BigInt(),
 		msg.Data.Payload,
 		chainID, uint(0), uint(0),
 	})
 }
 
 // EncodeRLP implements the rlp.Encoder interface.
 func (msg *MsgEthereumTx) EncodeRLP(w io.Writer) error {
-	return rlp.Encode(w, &msg.Data)
+	var hash ethcmn.Hash
+	if len(msg.Data.Hash) > 0 {
+		hash = ethcmn.HexToHash(msg.Data.Hash)
+	}
+
+	data := struct {
+		AccountNonce uint64
+		Price        *big.Int        `json:"gasPrice"`
+		GasLimit     uint64          `json:"gas"`
+		Recipient    *ethcmn.Address `json:"to" rlp:"nil"` // nil means contract creation
+		Amount       *big.Int        `json:"value"`
+		Payload      []byte          `json:"input"`
+
+		// signature values
+		V *big.Int `json:"v"`
+		R *big.Int `json:"r"`
+		S *big.Int `json:"s"`
```

Comment:
same for these, why are these now `[]byte` in the msg?

Reply:
there's no `big.Int` on protobuf